src/bin/trig.rs

```diff
@@ -424,32 +474,58 @@ impl SynthLanguage for Math {
                     continue;
                 };
 
-                if allowedp("+") {
-                    to_add.push(Math::Add([i, j]));
-                }
-                // if allowedp("-") { to_add.push(Math::RSub([i, j])); }
-                if allowedp("*") {
-                    to_add.push(Math::Mul([i, j]));
+                if iter <= 3 {
+                    // either both are trig or neither are trig
+                    if trig_node(&synth.egraph[i].nodes, &synth.egraph)
+                        == trig_node(&synth.egraph[j].nodes, &synth.egraph)
+                    {
+                        if allowedp("+") {
+                            to_add.push(Math::Add([i, j]));
+                        }
+
+                        if allowedp("-") {
+                            to_add.push(Math::Sub([i, j]));
+                        }
+
+                        if allowedp("*") {
+                            to_add.push(Math::Mul([i, j]));
+                        }
+                    }
+                } else if trig_node(&synth.egraph[i].nodes, &synth.egraph)
+                    && trig_node(&synth.egraph[j].nodes, &synth.egraph)
+                {
+                    if allowedp("+") {
+                        to_add.push(Math::Add([i, j]));
+                    }
+
+                    if allowedp("-") {
+                        to_add.push(Math::Sub([i, j]));
+                    }
                 }
-                // if allowedp("/") && !synth.egraph[j].nodes.iter().any(|x| is_zero(x)) {
-                //     to_add.push(Math::RDiv([i, j]));
-                // }
             }
 
-            if ids[&i] + 1 != iter || synth.egraph[i].data.exact || !synth.egraph[i].data.is_allowed
-            {
+            if ids[&i] + 1 != iter || synth.egraph[i].data.exact {
                 continue;
             }
 
-            // if allowedp("~") { to_add.push(Math::RNeg(i)); }
-            if iter <= 2 {
+            if iter <= 2 && arithmetic(&synth.egraph[i].nodes) {
```

Comment:
so is all this for your custom enumeration strategy?

Reply:
with enumo, should these go away?

Reply:
Eventually, built-in enumeration is not necessary. Might be nice to leave in for now.